___
<img style="float: right; margin: 0px 0px 15px 15px;" src="https://www.carrerasenlinea.mx/logos/original/logo-instituto-tecnologico-y-de-estudios-superiores-de-occidente.png" width="150px" height="100px" />


# MICROESTRUCTURAS Y SISTEMAS DE "TRADING"
# LABORATORIO 3

## Nombre: Jocelyn Medel Ruiz, Esteban Navarrete, Juan Lopez

## Fecha: 27 de Octubre 2019



## Disposition Effect
Descripción: Cortar más rapido las pérdidas que las ganancias.

Los inversionistas tienden a vender activos que aumentaron su valor y tienden a mantener los activos cuyo valor ha disminuido.

Analizar las operaciónes a las que pierdes y en las que ganas por sus lapsos de tiempo, revisar cuanto tiempo tomó tenerlas en perdidas y lo mismo con las de las ganancias.

Checar cuando cortó el TP/SL y que tan rápido lo cortaba según iba ganando o perdiendo. Sacar interpretaciones.

Aplicación en caso de trading: De un archivo tradeview se analizan los movimientos realizados, tomando en consideración los trades que fueron cerrados manualmente y los que cerraban al alcanzar el stop loss o take profit.

### Código

#### Se importan librerias necesarias para el funcionamiento correcto del codigo

In [16]:
import numpy as np
import pandas as pd
from datetime import datetime
import matplotlib.pyplot as plt
import numpy as np
import plotly.express as px 
import plotly.graph_objects as go #Libreria necesaria para usar plotly
# import json

#### Opciones para visualizar data frames en consola

In [17]:
pd.set_option('display.max_rows', 5000)
pd.set_option('display.max_columns', 500)
pd.set_option('display.width', 1000)

#### Ingreso de datos de entrada: leer archivo json

In [18]:
archivo = '../MyST-Lab3-Equipo4/archivo_tradeview_2.xlsx'
df_datos = pd.read_excel(archivo,
                        header=0,
                        sep=',',
                        parse_dates=False,
                        skip_blank_lines=True)
df_datos.head()

Order             openTime Type  Size  Symbol   openPrice       S/L        T/P            closeTime  closePrice  Commission  Taxes  Swap  Profit
0  24316034  2019.08.27 06:57:50  buy  0.04  xauusd  1529.21000  1523.000  1543.0000  2019.08.27 09:16:01  1533.69000         0.0      0     0   17.92
1  24334215  2019.08.27 17:06:20  buy  0.01  eurjpy   117.66400   116.950   118.7000  2019.08.29 06:38:16   117.37400         0.0      0     0   -2.73
2  24342155  2019.08.27 19:25:39  buy  0.05  eurjpy   117.34600   116.950   119.7000  2019.08.28 21:01:24   117.45200         0.0      0     0    5.00
3  24373260  2019.08.28 17:11:33  buy  0.01  eurusd     1.10791     1.104     1.1175  2019.08.29 06:38:28     1.10818         0.0      0     0    0.27
4  24392415  2019.08.29 06:35:27  buy  0.01  gbpusd     1.22024     1.208     1.2370  2019.08.29 06:37:23     1.21990         0.0      0     0   -0.34

In [19]:
df_datos = df_datos.drop(['Order', 'Size', 'Commission', 'Taxes', 'Swap'], 1) #El 1 se refiere a columnas, si fuera 0 serian filas

In [20]:
df_datos = df_datos.reset_index()

#### Renombramos las columnas en minusculas por convección

In [21]:
df_datos.rename(columns={'Type' : 'type', 'Symbol': 'symbol',
                            'S/L' : 'sl', 'T/P':'tp', 'Profit': 'profit'},
                inplace=True)
df_datos

index             openTime  type  symbol   openPrice         sl         tp            closeTime  closePrice  profit
0       0  2019.08.27 06:57:50   buy  xauusd  1529.21000  1523.0000  1543.0000  2019.08.27 09:16:01  1533.69000   17.92
1       1  2019.08.27 17:06:20   buy  eurjpy   117.66400   116.9500   118.7000  2019.08.29 06:38:16   117.37400   -2.73
2       2  2019.08.27 19:25:39   buy  eurjpy   117.34600   116.9500   119.7000  2019.08.28 21:01:24   117.45200    5.00
3       3  2019.08.28 17:11:33   buy  eurusd     1.10791     1.1040     1.1175  2019.08.29 06:38:28     1.10818    0.27
4       4  2019.08.29 06:35:27   buy  gbpusd     1.22024     1.2080     1.2370  2019.08.29 06:37:23     1.21990   -0.34
5       5  2019.08.29 06:36:15   buy  gbpjpy   129.23400   128.0000   132.2000  2019.08.29 06:36:58   129.21000   -0.23
6       6  2019.08.29 06:36:15   buy  gbpjpy   129.23400   128.0000   132.2000  2019.08.29 15:01:24   129.80100   21.35
7       7  2019.08.29 06:35:27   buy  gbpusd     1.22024     1.2080     1.2370  2019.08.29 15:01:43     1.22173    5.96
8       8  2019.08.27 17:06:20   buy  eurjpy   117.66400   117.4000   118.7000  2019.08.30 09:17:23   117.39900   -9.98
9       9  2019.08.28 17:11:33   buy  eurusd     1.10791     1.1015     1.1175  2019.08.30 17:57:15     1.10150  -25.64
10     10  2019.08.29 19:56:26   buy  eurusd     1.10543     1.1040     0.0000  2019.08.29 19:56:51     1.10518   -1.00
11     11  2019.08.29 19:57:00   buy  eurusd     1.10545     1.1015     0.0000  2019.08.30 17:57:14     1.10150  -19.75
12     12  2019.08.29 20:33:00   buy  xauusd  1528.76000  1524.0000  1555.0000  2019.08.29 21:11:08  1523.80000  -24.80
13     13  2019.08.30 06:55:31   buy  xauusd  1525.05000  1515.0000  1560.0000  2019.08.30 17:02:21  1528.61000   17.80
14     14  2019.09.02 01:33:23   buy  gbpusd     1.21630     1.2080     0.0000  2019.09.02 01:33:50     1.21590   -0.40
15     15  2019.09.02 01:33:23   buy  gbpusd     1.21630     1.2080     0.0000  2019.09.02 11:55:28     1.20799  -33.24
16     16  2019.09.02 06:58:46   buy  eurusd     1.09915     1.0940     0.0000  2019.09.03 04:04:43     1.09399  -25.80
17     17  2019.09.03 04:49:15   buy  gbpjpy   128.17800   127.8000     0.0000  2019.09.03 04:49:22   128.14500   -0.31
18     18  2019.09.03 04:49:15   buy  gbpjpy   128.17800   127.8000     0.0000  2019.09.03 08:37:26   127.80000  -14.24
19     19  2019.09.03 04:49:42   buy  audusd     0.67020     0.6660     0.0000  2019.09.03 04:49:54     0.66991   -0.29
20     20  2019.09.03 04:49:42   buy  audusd     0.67020     0.6660     0.6830  2019.09.03 15:47:53     0.67284   10.56
21     21  2019.09.03 04:50:18   buy  usdjpy   106.37200   105.8000     0.0000  2019.09.03 04:50:27   106.34500   -0.25
22     22  2019.09.03 04:50:18   buy  usdjpy   106.37200   105.8000     0.0000  2019.09.03 17:04:26   105.80000  -21.63
23     23  2019.09.11 01:52:07   buy  xauusd  1487.63000  1473.0000     0.0000  2019.09.11 01:52:25  1487.05000   -0.58
24     24  2019.09.11 01:52:07   buy  xauusd  1487.63000  1473.0000     0.0000  2019.09.11 14:11:29  1492.74000   20.44
25     25  2019.09.11 14:33:24   buy  usdmxn    19.52363    19.4300     0.0000  2019.09.12 14:48:03    19.42847  -29.39
26     26  2019.09.13 00:14:20   buy  usdmxn    19.44123     0.0000     0.0000  2019.09.25 15:24:28    19.58155   43.00
27     27  2019.09.13 05:27:30   buy  xauusd  1497.97000  1491.0000     0.0000  2019.09.13 10:58:49  1504.48000   39.06
28     28  2019.09.13 18:32:01   buy  xauusd  1494.14000     0.0000     0.0000  2019.09.16 01:24:24  1501.74000   45.60
29     29  2019.09.13 22:01:23   buy  xauusd  1487.17000     0.0000     0.0000  2019.09.13 23:48:34  1488.57000    8.40
30     30  2019.09.17 05:35:53   buy  eurusd     1.10122     1.0975     0.0000  2019.09.17 15:26:48     1.10214    5.52
31     31  2019.09.19 06:15:39   buy  eurusd     1.10373     0.0000     0.0000  2019.09.19 15:33:37     1.10639   15.96
32     32  2019.09.19 16:00:58   buy  usdmxn

In [22]:
df_times= df_datos[['openTime','closeTime']]
df_times

openTime            closeTime
0   2019.08.27 06:57:50  2019.08.27 09:16:01
1   2019.08.27 17:06:20  2019.08.29 06:38:16
2   2019.08.27 19:25:39  2019.08.28 21:01:24
3   2019.08.28 17:11:33  2019.08.29 06:38:28
4   2019.08.29 06:35:27  2019.08.29 06:37:23
5   2019.08.29 06:36:15  2019.08.29 06:36:58
6   2019.08.29 06:36:15  2019.08.29 15:01:24
7   2019.08.29 06:35:27  2019.08.29 15:01:43
8   2019.08.27 17:06:20  2019.08.30 09:17:23
9   2019.08.28 17:11:33  2019.08.30 17:57:15
10  2019.08.29 19:56:26  2019.08.29 19:56:51
11  2019.08.29 19:57:00  2019.08.30 17:57:14
12  2019.08.29 20:33:00  2019.08.29 21:11:08
13  2019.08.30 06:55:31  2019.08.30 17:02:21
14  2019.09.02 01:33:23  2019.09.02 01:33:50
15  2019.09.02 01:33:23  2019.09.02 11:55:28
16  2019.09.02 06:58:46  2019.09.03 04:04:43
17  2019.09.03 04:49:15  2019.09.03 04:49:22
18  2019.09.03 04:49:15  2019.09.03 08:37:26
19  2019.09.03 04:49:42  2019.09.03 04:49:54
20  2019.09.03 04:49:42  2019.09.03 15:47:53
21  2019.09.03 04:50:18  2019.09.03 04:50:27
22  2019.09.03 04:50:18  2019.09.03 17:04:26
23  2019.09.11 01:52:07  2019.09.11 01:52:25
24  2019.09.11 01:52:07  2019.09.11 14:11:29
25  2019.09.11 14:33:24  2019.09.12 14:48:03
26  2019.09.13 00:14:20  2019.09.25 15:24:28
27  2019.09.13 05:27:30  2019.09.13 10:58:49
28  2019.09.13 18:32:01  2019.09.16 01:24:24
29  2019.09.13 22:01:23  2019.09.13 23:48:34
30  2019.09.17 05:35:53  2019.09.17 15:26:48
31  2019.09.19 06:15:39  2019.09.19 15:33:37
32  2019.09.19 16:00:58  2019.09.19 17:53:54
33  2019.09.19 23:03:33  2019.09.20 07:27:14
34  2019.09.19 23:04:41  2019.09.20 07:27:10
35  2019.09.25 18:22:33  2019.09.26 21:00:44
36  2019.09.25 18:23:20  2019.09.26 21:00:44
37  2019.09.25 20:05:46  2019.09.26 22:33:46
38  2019.09.26 20:20:58  2019.09.27 15:49:19
39  2019.08.27 06:18:03  2019.08.27 06:20:11
40  2019.08.27 06:22:44  2019.08.27 06:23:28
41  2019.08.27 06:23:44  2019.08.27 06:24:13
42  2019.08.27 06:45:16  2019.08.27 09:16:08
43  2019.08.27 06:52:01  2019.08.27 09:16:04
44  2019.08.27 09:31:20  2019.08.27 13:31:11
45  2019.08.27 13:56:24  2019.08.27 13:57:03
46  2019.08.27 13:57:22  2019.08.27 13:57:57
47  2019.08.27 13:58:20  2019.08.27 14:02:05
48  2019.08.30 02:42:53  2019.08.30 02:53:33
49  2019.08.30 02:42:53  2019.08.30 17:02:18
50  2019.09.02 01:33:09  2019.09.02 01:33:56
51  2019.09.02 01:33:09  2019.09.03 15:20:26
52  2019.09.03 04:50:57  2019.09.03 16:23:12
53  2019.09.04 04:10:47  2019.09.04 17:00:25
54  2019.09.05 20:48:18  2019.09.06 13:50:46
55  2019.09.05 20:48:35  2019.09.06 13:50:41
56  2019.09.05 20:56:36  2019.09.06 13:50:38
57  2019.09.06 06:05:00  2019.09.06 06:05:20
58  2019.09.06 06:05:38  2019.09.06 06:05:52
59  2019.09.06 06:06:00  2019.09.06 13:50:35
60  2019.09.06 14:01:24  2019.09.06 15:30:49
61  2019.09.06 14:01:52  2019.09.06 15:30:53
62  2019.09.11 01:54:34  2019.09.11 01:54:40
63  2019.09.11 01:54:34  2019.09.11 14:11:23
64  2019.09.11 07:06:14  2019.09.11 14:11:19
65  2019.09.11 14:26:37  2019.09.11 17:18:40
66  2019.09.12 05:40:12  2019.09.12 15:48:51
67  2019.09.13 13:43:31  2019.09.13 16:47:28
68  2019.09.16 07:42:05  2019.09.16 20:34:11
69  2019.09.16 07:44:47  2019.09.16 14:52:43
70  2019.09.16 07:51:17  2019.09.16 14:49:38
71  2019.09.16 15:08:18  2019.09.16 20:34:08
72  2019.09.17 15:39:46  2019.09.17 16:17:43
73  2019.09.17 18:31:58  2019.09.18 21:16:55
74  2019.09.19 08:38:40  2019.09.19 15:41:41
75  2019.09.24 01:09:37  2019.09.24 01:10:35
76  2019.09.24 01:10:44  2019.09.24 01:12:25
77  2019.09.24 01:10:44  2019.09.24 17:01:38
78  2019.09.24 01:13:04  2019.09.24 17:01:38
79  2019.09.24 20:09:52  2019.09.24 21:18:43
80  2019.09.24 20:11:20  2019.09.24 21:18:46
81  2019.09.24 20:12:41  2019.09.24 21:18:49
82  2019.09.25 01:53:47  2019.09.25 15:24:31
83  2019.09.25 01:54:46  2019.09.25 15:24:35

In [23]:
df_times['openTime'] = df_times['openTime'].str.replace('.','-')
df_times['closeTime'] = df_times['closeTime'].str.replace('.','-')
df_times

//anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:1: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy

//anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:2: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy



openTime            closeTime
0   2019-08-27 06:57:50  2019-08-27 09:16:01
1   2019-08-27 17:06:20  2019-08-29 06:38:16
2   2019-08-27 19:25:39  2019-08-28 21:01:24
3   2019-08-28 17:11:33  2019-08-29 06:38:28
4   2019-08-29 06:35:27  2019-08-29 06:37:23
5   2019-08-29 06:36:15  2019-08-29 06:36:58
6   2019-08-29 06:36:15  2019-08-29 15:01:24
7   2019-08-29 06:35:27  2019-08-29 15:01:43
8   2019-08-27 17:06:20  2019-08-30 09:17:23
9   2019-08-28 17:11:33  2019-08-30 17:57:15
10  2019-08-29 19:56:26  2019-08-29 19:56:51
11  2019-08-29 19:57:00  2019-08-30 17:57:14
12  2019-08-29 20:33:00  2019-08-29 21:11:08
13  2019-08-30 06:55:31  2019-08-30 17:02:21
14  2019-09-02 01:33:23  2019-09-02 01:33:50
15  2019-09-02 01:33:23  2019-09-02 11:55:28
16  2019-09-02 06:58:46  2019-09-03 04:04:43
17  2019-09-03 04:49:15  2019-09-03 04:49:22
18  2019-09-03 04:49:15  2019-09-03 08:37:26
19  2019-09-03 04:49:42  2019-09-03 04:49:54
20  2019-09-03 04:49:42  2019-09-03 15:47:53
21  2019-09-03 04:50:18  2019-09-03 04:50:27
22  2019-09-03 04:50:18  2019-09-03 17:04:26
23  2019-09-11 01:52:07  2019-09-11 01:52:25
24  2019-09-11 01:52:07  2019-09-11 14:11:29
25  2019-09-11 14:33:24  2019-09-12 14:48:03
26  2019-09-13 00:14:20  2019-09-25 15:24:28
27  2019-09-13 05:27:30  2019-09-13 10:58:49
28  2019-09-13 18:32:01  2019-09-16 01:24:24
29  2019-09-13 22:01:23  2019-09-13 23:48:34
30  2019-09-17 05:35:53  2019-09-17 15:26:48
31  2019-09-19 06:15:39  2019-09-19 15:33:37
32  2019-09-19 16:00:58  2019-09-19 17:53:54
33  2019-09-19 23:03:33  2019-09-20 07:27:14
34  2019-09-19 23:04:41  2019-09-20 07:27:10
35  2019-09-25 18:22:33  2019-09-26 21:00:44
36  2019-09-25 18:23:20  2019-09-26 21:00:44
37  2019-09-25 20:05:46  2019-09-26 22:33:46
38  2019-09-26 20:20:58  2019-09-27 15:49:19
39  2019-08-27 06:18:03  2019-08-27 06:20:11
40  2019-08-27 06:22:44  2019-08-27 06:23:28
41  2019-08-27 06:23:44  2019-08-27 06:24:13
42  2019-08-27 06:45:16  2019-08-27 09:16:08
43  2019-08-27 06:52:01  2019-08-27 09:16:04
44  2019-08-27 09:31:20  2019-08-27 13:31:11
45  2019-08-27 13:56:24  2019-08-27 13:57:03
46  2019-08-27 13:57:22  2019-08-27 13:57:57
47  2019-08-27 13:58:20  2019-08-27 14:02:05
48  2019-08-30 02:42:53  2019-08-30 02:53:33
49  2019-08-30 02:42:53  2019-08-30 17:02:18
50  2019-09-02 01:33:09  2019-09-02 01:33:56
51  2019-09-02 01:33:09  2019-09-03 15:20:26
52  2019-09-03 04:50:57  2019-09-03 16:23:12
53  2019-09-04 04:10:47  2019-09-04 17:00:25
54  2019-09-05 20:48:18  2019-09-06 13:50:46
55  2019-09-05 20:48:35  2019-09-06 13:50:41
56  2019-09-05 20:56:36  2019-09-06 13:50:38
57  2019-09-06 06:05:00  2019-09-06 06:05:20
58  2019-09-06 06:05:38  2019-09-06 06:05:52
59  2019-09-06 06:06:00  2019-09-06 13:50:35
60  2019-09-06 14:01:24  2019-09-06 15:30:49
61  2019-09-06 14:01:52  2019-09-06 15:30:53
62  2019-09-11 01:54:34  2019-09-11 01:54:40
63  2019-09-11 01:54:34  2019-09-11 14:11:23
64  2019-09-11 07:06:14  2019-09-11 14:11:19
65  2019-09-11 14:26:37  2019-09-11 17:18:40
66  2019-09-12 05:40:12  2019-09-12 15:48:51
67  2019-09-13 13:43:31  2019-09-13 16:47:28
68  2019-09-16 07:42:05  2019-09-16 20:34:11
69  2019-09-16 07:44:47  2019-09-16 14:52:43
70  2019-09-16 07:51:17  2019-09-16 14:49:38
71  2019-09-16 15:08:18  2019-09-16 20:34:08
72  2019-09-17 15:39:46  2019-09-17 16:17:43
73  2019-09-17 18:31:58  2019-09-18 21:16:55
74  2019-09-19 08:38:40  2019-09-19 15:41:41
75  2019-09-24 01:09:37  2019-09-24 01:10:35
76  2019-09-24 01:10:44  2019-09-24 01:12:25
77  2019-09-24 01:10:44  2019-09-24 17:01:38
78  2019-09-24 01:13:04  2019-09-24 17:01:38
79  2019-09-24 20:09:52  2019-09-24 21:18:43
80  2019-09-24 20:11:20  2019-09-24 21:18:46
81  2019-09-24 20:12:41  2019-09-24 21:18:49
82  2019-09-25 01:53:47  2019-09-25 15:24:31
83  2019-09-25 01:54:46  2019-09-25 15:24:35

In [24]:
df_datos['openTime'] = df_times['openTime']
df_datos['closeTime'] = df_times['closeTime']
df_datos

index             openTime  type  symbol   openPrice         sl         tp            closeTime  closePrice  profit
0       0  2019-08-27 06:57:50   buy  xauusd  1529.21000  1523.0000  1543.0000  2019-08-27 09:16:01  1533.69000   17.92
1       1  2019-08-27 17:06:20   buy  eurjpy   117.66400   116.9500   118.7000  2019-08-29 06:38:16   117.37400   -2.73
2       2  2019-08-27 19:25:39   buy  eurjpy   117.34600   116.9500   119.7000  2019-08-28 21:01:24   117.45200    5.00
3       3  2019-08-28 17:11:33   buy  eurusd     1.10791     1.1040     1.1175  2019-08-29 06:38:28     1.10818    0.27
4       4  2019-08-29 06:35:27   buy  gbpusd     1.22024     1.2080     1.2370  2019-08-29 06:37:23     1.21990   -0.34
5       5  2019-08-29 06:36:15   buy  gbpjpy   129.23400   128.0000   132.2000  2019-08-29 06:36:58   129.21000   -0.23
6       6  2019-08-29 06:36:15   buy  gbpjpy   129.23400   128.0000   132.2000  2019-08-29 15:01:24   129.80100   21.35
7       7  2019-08-29 06:35:27   buy  gbpusd     1.22024     1.2080     1.2370  2019-08-29 15:01:43     1.22173    5.96
8       8  2019-08-27 17:06:20   buy  eurjpy   117.66400   117.4000   118.7000  2019-08-30 09:17:23   117.39900   -9.98
9       9  2019-08-28 17:11:33   buy  eurusd     1.10791     1.1015     1.1175  2019-08-30 17:57:15     1.10150  -25.64
10     10  2019-08-29 19:56:26   buy  eurusd     1.10543     1.1040     0.0000  2019-08-29 19:56:51     1.10518   -1.00
11     11  2019-08-29 19:57:00   buy  eurusd     1.10545     1.1015     0.0000  2019-08-30 17:57:14     1.10150  -19.75
12     12  2019-08-29 20:33:00   buy  xauusd  1528.76000  1524.0000  1555.0000  2019-08-29 21:11:08  1523.80000  -24.80
13     13  2019-08-30 06:55:31   buy  xauusd  1525.05000  1515.0000  1560.0000  2019-08-30 17:02:21  1528.61000   17.80
14     14  2019-09-02 01:33:23   buy  gbpusd     1.21630     1.2080     0.0000  2019-09-02 01:33:50     1.21590   -0.40
15     15  2019-09-02 01:33:23   buy  gbpusd     1.21630     1.2080     0.0000  2019-09-02 11:55:28     1.20799  -33.24
16     16  2019-09-02 06:58:46   buy  eurusd     1.09915     1.0940     0.0000  2019-09-03 04:04:43     1.09399  -25.80
17     17  2019-09-03 04:49:15   buy  gbpjpy   128.17800   127.8000     0.0000  2019-09-03 04:49:22   128.14500   -0.31
18     18  2019-09-03 04:49:15   buy  gbpjpy   128.17800   127.8000     0.0000  2019-09-03 08:37:26   127.80000  -14.24
19     19  2019-09-03 04:49:42   buy  audusd     0.67020     0.6660     0.0000  2019-09-03 04:49:54     0.66991   -0.29
20     20  2019-09-03 04:49:42   buy  audusd     0.67020     0.6660     0.6830  2019-09-03 15:47:53     0.67284   10.56
21     21  2019-09-03 04:50:18   buy  usdjpy   106.37200   105.8000     0.0000  2019-09-03 04:50:27   106.34500   -0.25
22     22  2019-09-03 04:50:18   buy  usdjpy   106.37200   105.8000     0.0000  2019-09-03 17:04:26   105.80000  -21.63
23     23  2019-09-11 01:52:07   buy  xauusd  1487.63000  1473.0000     0.0000  2019-09-11 01:52:25  1487.05000   -0.58
24     24  2019-09-11 01:52:07   buy  xauusd  1487.63000  1473.0000     0.0000  2019-09-11 14:11:29  1492.74000   20.44
25     25  2019-09-11 14:33:24   buy  usdmxn    19.52363    19.4300     0.0000  2019-09-12 14:48:03    19.42847  -29.39
26     26  2019-09-13 00:14:20   buy  usdmxn    19.44123     0.0000     0.0000  2019-09-25 15:24:28    19.58155   43.00
27     27  2019-09-13 05:27:30   buy  xauusd  1497.97000  1491.0000     0.0000  2019-09-13 10:58:49  1504.48000   39.06
28     28  2019-09-13 18:32:01   buy  xauusd  1494.14000     0.0000     0.0000  2019-09-16 01:24:24  1501.74000   45.60
29     29  2019-09-13 22:01:23   buy  xauusd  1487.17000     0.0000     0.0000  2019-09-13 23:48:34  1488.57000    8.40
30     30  2019-09-17 05:35:53   buy  eurusd     1.10122     1.0975     0.0000  2019-09-17 15:26:48     1.10214    5.52
31     31  2019-09-19 06:15:39   buy  eurusd     1.10373     0.0000     0.0000  2019-09-19 15:33:37     1.10639   15.96
32     32  2019-09-19 16:00:58   buy  usdmxn

#### Se realiza una resta entre las fechas que abrio la operación y la cerro

In [25]:
#Ciclo para medir tiempo entre close time y open time manuales de tp y sl

manualTime_tp = []
manualTime_sl = []
for i, row in df_datos.iterrows():
    if row['profit'] > 0 and row['tp'] > 0:
        if row['closePrice']-row['tp'] != 0:
            calc_temp = (datetime.fromisoformat(df_datos.iloc[i,7]).timestamp()
            -datetime.fromisoformat(df_datos.iloc[i,1]).timestamp())/60/60
            manualTime_tp.append(calc_temp)
        else:
            pass
    elif row['profit'] < 0 and row['sl'] > 0:
        if row['closePrice']-row['sl'] != 0:
            calc_temp = (datetime.fromisoformat(df_datos.iloc[i,7]).timestamp()
            -datetime.fromisoformat(df_datos.iloc[i,1]).timestamp())/60/60
            manualTime_sl.append(calc_temp)
        else:
            pass

In [26]:
manualTime_tp = pd.DataFrame(manualTime_tp, columns=["manual_tp"])
manualTime_sl = pd.DataFrame(manualTime_sl, columns=["manual_sl"])



In [27]:
#Dataframe final
df_manualTimes = pd.concat([manualTime_tp, manualTime_sl], axis=1, sort=False)
df_manualTimes

manual_tp  manual_sl
0    2.303056  37.532222
1   25.595833   0.032222
2   13.448611   0.011944
3    8.419167  64.184167
4    8.437778   0.006944
5   10.113889   0.635556
6   10.969722   0.007500
7   26.636389  10.368056
8   26.623333  21.099167
9    2.514444   0.001944
10   3.997500   0.003333
11   1.490278   0.002500
12   7.050278   0.005000
13  15.848333  24.244167
14  15.809444  26.466667
15        NaN   0.035556
16        NaN   0.012222
17        NaN   2.400833
18        NaN   0.010833
19        NaN   0.009722
20        NaN   0.062500
21        NaN   0.013056
22        NaN  37.788056
23        NaN  17.035000
24        NaN   0.005556
25        NaN   0.003889
26        NaN   7.743056
27        NaN   1.483611
28        NaN   0.001667
29        NaN   0.632500
30        NaN   0.016111

#### Graficar los tiempos que se tardó, con cada uno de los trades.

#### Presentación de grafica: Box Plot

Los extremos de la caja representan los cuartiles inferior y superior, la mediana está marcada por una linea dentro de la caja.


In [28]:
fig = go.Figure()
fig.add_trace(go.Box(
    y=df_manualTimes.iloc[0:,1],
    name='Only Mean',
    marker_color='darkblue',
    boxmean=True # represent mean
))

fig.show()

### CASO GENÈRICO
#### Nuestro analisis fue enfocado en cortar más rapido las pérdidas que las ganancias, en un caso en particular de nuestra lista, el Trade 1, EurJpy se perdió un total de 2.73 después de dos dias que se abrió la transacción, esta fue abierta en 117.66 yenes por cada Euro, la orden era de compra por o que se esperaba que el Euro subierá respecto al Yen Japonés, la orden fue abierta el 27 de Agosto del 2019 a las 5 de la tarde, y fue cerrada a las 6pm dos dias después el 29 a las 6:38 de la madrugada, con un precio de 117.37. Se fijó un precio de 116.950, 118.7000, para el stop loss y el take profit; es de notar que no alcanzó ninguno de los limites referidos, la transacción fue cerrada manualmente. En un caso como este se podría fijar un stop loss más corto para lograr tener una mayor, perdida, existen otros casos en los cuales se intenta recuperar, pero existe una mayor perdida, este podría probar la situaciónen la que a la poca perdida cerrar el trade, aúnque no sabemos si en el curso del tiempo llegó a tener una mayor perdida.


## Conclusión
### En nuestro disposition Effect nuestro punto fue llegar al análisis de las operaciónes a las que pierdes y en las que ganas por sus lapzos de tiempo, revisar cuanto tiempo tomó tenerlas en perdidas y lo mismo con las de las ganancias, nuestros resultados demuestran el tiempo en que el Trade estuvo en ganancia, el tiempo que estuvo en perdida, por lo que en nuestro código se muestra como manual_tp y manual_sl traducidos al cierre de operación, en stop loss y en take profit de los cuales obtuvimos 30 datos de los 85 trades ejecutados. A partir del 15vo dato en delante, nos damos cuenta que los resultados en tp es decir take profit aparecen con el dato de Nan, lo que nos habla de como la columna del manual_tp, posee mayores datos los cuales no aparecen en las celdas vacías y python los rellena con Nan, se puede decir con esto que 15 veces el trader paró manualmente las operanciones que iban con ganancia y 31 veces las que iban con perdida. Mientras que en la gráfica visualmente podemos apreciar como existen mayor tiempo en los que hubo perdidas a diferencia de las ganancias, se podría decir que se esperaba la recuperación.
### Con ello y con el objetivo principal de obtener menores perdidas en los futuros trades podemos concluir que, con un buen analisis de los datos podemos obtener mejores resultados, mayores ganancias en el total de transacciónes que se realizen.